contribute/design/1.x/apis/cstorvolume_policy.md

```diff
@@ -0,0 +1,320 @@
+	Message string `json:"message"`
+}
+
+// CStorVolumeConfigConditionType is a valid value of CstorVolumeConfigCondition.Type
+type CStorVolumeConfigConditionType string
+
+// These constants are CVC condition types related to resize operation.
+const (
+	// CStorVolumeConfigResizePending means volume resize operation has been started
+	CStorVolumeConfigResizing CStorVolumeConfigConditionType = "Resizing"
+
+  // CStorVolumeConfigResizeFailed means volume resize operation has been failed
+	CStorVolumeConfigResizeFailed CStorVolumeConfigConditionType = "VolumeResizeFailed"
+	
+  // CStorVolumeConfigResizeSuccess means volume resize operation has been successful
+	CStorVolumeConfigResizeSuccess CStorVolumeConfigConditionType = "VolumeResizeSuccessful"
+	
+  // CStorVolumeConfigResizePending means volume resize operation has been pending
+	CStorVolumeConfigResizePending CStorVolumeConfigConditionType = "VolumeResizePending"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// CStorVolumeConfigList is a list of CStorVolumeConfig resources
+type CStorVolumeConfigList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []CStorVolumeConfig `json:"items"`
+}
+```
+
+# Changes to existing types
+
+- No changes
+
+# Volume Policy Schema Proposal
+
+## `CStorVolumePolicy` APIs
+
+```go
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +k8s:openapi-gen=true
+
+// CStorVolumePolicy describes a configuration required for cstor volume
+// resources
+type CStorVolumePolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec defines a configuration info of a cstor volume required
+	// to provisione cstor volume resources
+	Spec   CStorVolumePolicySpec   `json:"spec"`
+	Status CStorVolumePolicyStatus `json:"status"`
+}
+
+// CStorVolumePolicySpec ...
+type CStorVolumePolicySpec struct {
+  // Provision represents volume provisioning configuration like replicaAffinity
+  // etc
+	Provision Provision `json:"provision"`
+
+  // TargetSpec represents configuration related to cstor target and its resources
+	Target TargetSpec `json:"target,omitempty"`
+
+  // ReplicaSpec represents configuration related to replicas resources
+	Replica ReplicaSpec `json:"replica,omitempty"`
+
+
+	// ReplicaPoolInfo holds the pool information of volume replicas.
+	// Ex: If volume is provisioned on which CStor pool volume replicas exist
+	ReplicaPoolInfo []ReplicaPoolInfo `json:"replicaPoolInfo,omitempty"`
+}
+
+// TargetSpec represents configuration related to cstor target and its resources
+type TargetSpec struct {
+	// QueueDepth sets the queue size at iSCSI target which limits the
+	// ongoing IO count from client
+	QueueDepth string `json:"queueDepth,omitempty"`
+
+	// Luworkers sets the number of threads that are working on above queue
+	LuWorkers int64 `json:"luWorkers,omitempty"`
```

Comment:
n_: LuWorkers to IOThreads?

Reply:
actually both `Zvolworkers` and `Luworkers` are IO workers for zfs and istgt respectively , how can we differentiate ?

Reply:
this being in target spec, it is specific to target i.e., istgt @prateekpandey14 .. so, we are fine in changing its name

Reply:
Changed to `IOWorkers`